tests/system/python/packages/test_authentication.py

```diff
@@ -0,0 +1,1157 @@
+        conn.request("POST", "/foglamp/login", json.dumps({"username": "admin", "password": "foglamp"}))
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert "Logged in successfully" == jdoc['message']
+        _token = jdoc["token"]
+
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("PUT", '/foglamp/category/rest_api', body=json.dumps({"allowPing": "false"}),
+                     headers={"authorization": _token})
+        r = conn.getresponse()
+        assert 200 == r.status
+
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("PUT", '/foglamp/logout', headers={"authorization": _token})
+        r = conn.getresponse()
+        assert 200 == r.status
+
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("GET", "/foglamp/ping")
+        r = conn.getresponse()
+        assert 403 == r.status
+        assert "Forbidden" == r.reason
+
+    def test_ping_with_allowping_false_with_certificate_token(self, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
+        cert_file_path = os.path.join(os.path.expandvars('${FOGLAMP_ROOT}'), 'data/etc/certs/admin.cert')
+        with open(cert_file_path, 'r') as f:
+            conn.request("POST", "/foglamp/login", body=f)
+            r = conn.getresponse()
+            assert 200 == r.status
+            r = r.read().decode()
+            jdoc = json.loads(r)
+            assert "Logged in successfully" == jdoc['message']
+            _token = jdoc["token"]
+
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("PUT", '/foglamp/category/rest_api', json.dumps({"allowPing": "false"}),
+                     headers={"authorization": _token})
+        r = conn.getresponse()
+        assert 200 == r.status
+
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("PUT", '/foglamp/logout', headers={"authorization": _token})
+        r = conn.getresponse()
+        assert 200 == r.status
+
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("GET", "/foglamp/ping")
+        r = conn.getresponse()
+        assert 403 == r.status
+        assert "Forbidden" == r.reason
+
+    @pytest.mark.parametrize(("query", "expected_values"), [
+        ('', {'users': [{'userId': 1, 'roleId': 1, 'userName': 'admin'},
+                        {'userId': 2, 'roleId': 2, 'userName': 'user'}]}),
+        ('?id=2', {'userId': 2, 'roleId': 2, 'userName': 'user'}),
+        ('?username=admin', {'userId': 1, 'roleId': 1, 'userName': 'admin'}),
+        ('?id=1&username=admin', {'userId': 1, 'roleId': 1, 'userName': 'admin'}),
+    ])
+    def test_get_users_with_password_token(self, foglamp_url, query, expected_values):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("GET", "/foglamp/user{}".format(query), headers={"authorization": PASSWORD_TOKEN})
```

Comment:
from where we got this `PASSWORD_TOKEN` I see some are with global, some in test itself?